### PR TITLE
[#374] Validate format constraints on string properties as described …

### PR DESCRIPTION
**Pull request by ghillairet**
…in OpenAPI 3.0 spec

See #374 

Modified schema to validate email property

### CODE REVIEW

**Comment by ghillairet** [comment]
@tfesenko See this PR, it includes changes in the schema to allow validation of email properties. I haven't found how to validate url properties yet, maybe we will need to to it ourselves.

media2

File: com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/schema/schema_v3.json [hunk 1]
```diff
@@ -105,7 +105,8 @@
           "type": "string"
         },
         "email": {
-          "type": "string"
+          "type": "string",
+          "format": "email"
```
Comment (tfesenko) [inline com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/schema/schema_v3.json:110]: 👍

**Comment by ghillairet** [comment]
@tfesenko The v2 schema already defines the correct format for those properties (see https://github.com/RepreZen/KaiZen-OpenAPI-Editor/blob/master/com.reprezen.swagedit/src/com/reprezen/swagedit/schema/schema.json#L129) The email validation is working. The uri validation does not work in the editor but works when I run tests on it. I'm trying to understand why it does not show up in the editor.

**Comment by ghillairet** [comment]
The validation done by the JSON schema validator already uses java URI to check the validity of the URI but the results may not be what is expected, see

media3

That can be reproduced in current v2 editor.

With changes I made to implement this functionality myself, I get the same results.

media4

**Comment by tfesenko** [comment]
See https://github.com/googleapis/gnostic/pull/63

**Pull request merged**